src/kernels/kernelFinder.base.ts

```diff
@@ -51,22 +52,17 @@ export abstract class BaseKernelFinder implements IKernelFinder {
     public async rankKernels(
         resource: Resource,
         notebookMetadata?: nbformat.INotebookMetadata,
+        preferredInterpreter?: PythonEnvironment,
         cancelToken?: CancellationToken,
-        useCache?: 'useCache' | 'ignoreCache'
+        useCache?: 'useCache' | 'ignoreCache',
+        serverId?: string
     ): Promise<KernelConnectionMetadata[] | undefined> {
-        const resourceType = getResourceType(resource);
         try {
             // Get list of all of the specs from the cache and without the cache (note, cached items will be validated before being returned)
-            const kernels = await this.listKernels(resource, cancelToken, useCache);
-
-            const isPythonNbOrInteractiveWindow = isPythonNotebook(notebookMetadata) || resourceType === 'interactive';
-
-            // Always include the interpreter in the search if we can
-            const preferredInterpreter =
```

Comment:
Removed this code from the base method, the exact same code `preferredInterpreter` is calculated in the calling code, hence might as well pass that into this method instead of computing this here, also remove a few dependencies.